### PR TITLE
new intresting features

### PR DESCRIPTION
**Pull request by AttilioGreco**
- Improve consul check
- HA mode
- Openstack CloudConfig
- Add Rancher Configurations

### CODE REVIEW

File: tasks/configure-bootstrap.yml [hunk 1]
```diff
@@ -72,3 +72,19 @@
       [k8s-cluster:children]
       kube-master
       kube-node
+# record remove in master-postconf
```
Comment (jsecchiero) [inline tasks/configure-bootstrap.yml]: add empty line
modify "record remove" in "record to remove"

File: tasks/consul.yml [hunk 2]
```diff
@@ -23,35 +23,69 @@
...
+        check:
+          - tcp: localhost:6443
+            interval: 10s
+            timeout: 3s
+      - name: "{{ cluster_name }}-kube-schedule"
+        check:
+          - tcp: localhost:10251
+            interval: 10s
+            timeout: 3s
+      - name: "{{ cluster_name }}-kube-controll"
```
Comment (jsecchiero) [inline tasks/consul.yml]: *controller

File: tasks/consul.yml [hunk 3]
```diff
@@ -23,35 +23,69 @@
...
       {% endif %}
-      {% if node == 'true' %}
-      - name: "{{ cluster_name }}-kube-node"
+      {% if node == 'true' or master == 'true' %}
       - name: "{{ cluster_name }}-kubelet"
         check:
           - tcp: localhost:10250
             interval: 10s
             timeout: 1s
+      - name: "{{ cluster_name }}-kubelet-health"
```
Comment (jsecchiero) [inline tasks/consul.yml:74]: add empty line between tasks

File: tasks/consul.yml [hunk 4]
```diff
@@ -23,35 +23,69 @@
...
+        check:
+          - http: http://localhost:10248/healthz
+            method: "GET"
+            interval: 30s
+            timeout: 5s
```
Comment (jsecchiero) [inline tasks/consul.yml:79]: add empty line between tasks

File: tasks/post-conf-master-etcd.yml [hunk 5]
```diff
@@ -1,23 +1,23 @@
 ---
-- name: post-conf | grant access to folder /etc/ssl/etcd consul user
+- name: post-conf-master-etcd | grant access to folder /etc/ssl/etcd consul user
```
Comment (jsecchiero) [inline tasks/post-conf-master-etcd.yml:3]: add empty line between tasks and ---

File: tasks/post-conf-master-etcd.yml [hunk 6]
```diff
@@ -1,23 +1,23 @@
...
   acl:
     path: /etc/ssl/etcd
     entity: consul
     etype: user
     permissions: rx
     state: present
-- name: post-conf | grant access to folder /etc/ssl/etcd/ssl consul user
+- name: post-conf-master-etcd | grant access to folder /etc/ssl/etcd/ssl consul user
```
Comment (jsecchiero) [inline tasks/post-conf-master-etcd.yml:11]: add empty line between tasks

File: tasks/post-conf-master-etcd.yml [hunk 7]
```diff
@@ -1,23 +1,23 @@
...
   acl:
     path: /etc/ssl/etcd/ssl
     entity: consul
     etype: user
     permissions: rx
     state: present
-- name: post-conf | set clustername
+- name: post-conf-master-etcd | set clustername
```
Comment (jsecchiero) [inline tasks/post-conf-master-etcd.yml:19]: add empty line between tasks

File: tasks/post-conf-master-etcd.yml [hunk 8]
```diff
@@ -1,23 +1,23 @@
...
   set_fact:
     cluster_name: "{{ cluster_name }}"
 
-- name: post-conf | grant access multiple file
+- name: post-conf-master-etcd | grant access multiple file
```
Comment (jsecchiero) [inline tasks/post-conf-master-etcd.yml:23]: add empty line between tasks

File: tasks/post-conf-master-etcd.yml [hunk 9]
```diff
@@ -29,33 +29,69 @@
...
-- name: post-conf | grant access to folder /root/
+- name: post-conf-master-etcd | grant access to folder /root/
   acl:
     path: /root
     entity: consul
     etype: user
     permissions: rx
     state: present
-- name: post-conf | grant access to folder /root/.kube
+- name: post-conf-master-etcd | grant access to folder /root/.kube
```
Comment (jsecchiero) [inline tasks/post-conf-master-etcd.yml:43]: add empty line between tasks

File: tasks/post-conf-master-etcd.yml [hunk 10]
```diff
@@ -29,33 +29,69 @@
...
   acl:
     path: /root/.kube
     entity: consul
     etype: user
     permissions: rx
     state: present
-- name: post-conf | grant access to folder /root/.kube/config
+- name: post-conf-master-etcd | grant access to folder /root/.kube/config
```
Comment (jsecchiero) [inline tasks/post-conf-master-etcd.yml:51]: add empty line between tasks

File: tasks/post-conf-master-etcd.yml [hunk 11]
```diff
@@ -29,33 +29,69 @@
...
   acl:
     path: "/root/.kube/config"
     entity: consul
     etype: user
     permissions: r
     recursive: yes
     state: present
-- name: post-conf | allow access to kubeconfig
+- name: post-conf-master-etcd | allow access to kubeconfig
```
Comment (jsecchiero) [inline tasks/post-conf-master-etcd.yml:60]: add empty line between tasks

File: tasks/post-conf-node.yml [hunk 12]
```diff
@@ -1,3 +1,12 @@
 ---
+- name: post-conf-node | remove kubernetes loadbalancer address into hosts file
```
Comment (jsecchiero) [inline tasks/post-conf-node.yml:3]: add empty line between tasks and ---

File: tasks/post-configure-bootstrap.yml [hunk 13]
```diff
@@ -0,0 +1,21 @@
+---
+- name: run script to add kubernetes to rancher
```
Comment (jsecchiero) [inline tasks/post-configure-bootstrap.yml:3]: add empty line between tasks and ---

File: tasks/copy-cert.yml [hunk 14]
```diff
@@ -16,11 +16,79 @@
...
+
+- name: copy-cert | configure host IP
+  set_fact:
+    loadbalancer_apiserver_address: >-
+      {% if master == "true" %}
+      {{ ip | default(ansible_default_ipv4.address) }}
+      {%- else -%}
+      {{ host_list_kube_master_up[0] }}
+      {%- endif -%}
+- name: set apiserver_loadbalancer_domain_name
```
Comment (jsecchiero) [inline tasks/copy-cert.yml:69]: add empty line between tasks and ---

File: tasks/configure-bootstrap.yml [hunk 15]
```diff
@@ -72,3 +72,19 @@
...
+- name: configure-bootstrap | Create "{{ cluster_name }}-etcd.service.automium.consul" record in /etc/hosts for consul boostrap
```
Comment (jsecchiero) [inline tasks/configure-bootstrap.yml]: start with lowecase letter

File: tasks/configure-bootstrap.yml [hunk 16]
```diff
@@ -72,3 +72,19 @@
...
+  lineinfile:
+    state: present
+    path: /etc/hosts
+    line: "{{ ansible_default_ipv4['address'] }} {{ cluster_name }}-etcd.service.automium.consul"
+  when:
+    - etcd == "true"
+
+- name: configure-bootstrap | Create "{{ cluster_name }}-kube-apiserver.service.automium.consul" record in /etc/hosts for consul boostrap
```
Comment (jsecchiero) [inline tasks/configure-bootstrap.yml]: start with lowecase letter

**Review by jsecchiero — CHANGES_REQUESTED** [review]
just some cosmetic fix before merge

File: tasks/consul.yml (hunk 2 above)
Comment (AttilioGreco) [reply tasks/consul.yml]: net stat truncate process name : (

FIX:
- kube-scheduler
- kube-controller-manager

File: tasks/consul.yml (hunk 3 above)
Comment (AttilioGreco) [reply tasks/consul.yml:74]: is part of 
- name: consul | render consul service
Start at line 34 end line 88 task name

File: tasks/consul.yml (hunk 4 above)
Comment (AttilioGreco) [reply tasks/consul.yml:79]: is part of 
- name: consul | render consul service
Start at line 34 end line 88 task name

**Review by jsecchiero — APPROVED** [review]

**Pull request merged**